datahost-ld-openapi/src/tpximpact/datahost/ldapi/util/data_compilation.clj

```diff
@@ -22,6 +31,74 @@
   (with-open [is (store/get-data store v)]
    (as-dataset is opts)))
 
+(defn make-columnwise-hasher []
+  (fn hasher* [& columns]
+    (let [^StringBuilder sb (StringBuilder.)]
+      (.append sb "|")
+      (loop [columns columns]
+        (if-not (seq columns)
+          (MurmurHash3/hash32 (.toString sb))
+          (do
+            (.append sb (first columns))
+            (.append sb "|")
+            (recur (next columns))))))))
+
+(def ^:private measure-type-uri (:measure uris/column-types))
+
+(defn- make-column-info-xform
+  [filter-fn]
+  (comp (map m/properties)
+        (filter filter-fn)
+        (map #(get % :dataset.column/name))))
+
+(def ^:private tuple-schema-component-names-xform
+  "Transducer transforming elements of [[m/children]] output into strings."
+  (make-column-info-xform #(not= measure-type-uri (get % :dataset.column/type))))
+
+(def ^:private tuple-schema-measure-names-xform
+  "Transducer transforming elements of [[mu/children]] output into strings."
+  (make-column-info-xform #(= measure-type-uri (get % :dataset.column/type))))
+
+(defn check-dataset-has-columns
```

Comment:
Is this something that could be handled by our existing validation logic? i.e. we already check that append and delete files conform the the schema, right? Could we run that same validation instead of doing this?